invenio_admin/ext.py

```diff
@@ -62,26 +62,46 @@ def register_view(self, view_class, model_class, session=None, **kwargs):
         :param kwargs: Passed to the ``view_class`` returned from the
             ``view_class_factory``.
         """
-        view_class = self.view_class_factory(view_class)
-        self.admin.add_view(
-            view_class(model_class, session or db.session, **kwargs))
+        protected_view_class = self.view_class_factory(view_class)
+        session = kwargs.pop('session') if 'session' in kwargs else None
+        # TODO: Backwards compatibility with old signature:
+        # register_view(self, view_class, model_class, session=None, **kwargs)
+        # We assume that there can be only one extra positional argument,
+        # at which point we register the ModelView, otherwise we register a
+        # custom admin view.
+        assert len(args) <= 1, \
```

Comment:
Please use `if` and `raise` instead of `assert` in normal code since `assert` statement can be disabled (`-O`).